content/docs/intro/languages/python.md

```diff
@@ -40,7 +44,7 @@ This will leave behind a `Pulumi.yaml` file, containing some minimal metadata ab
 
 ## Using Pulumi PyPI Packages {#pypi-packages}
 
-It is not required, but we recommend using [`pipenv`](https://pipenv.readthedocs.io/en/latest/) for repeatable installations that work no matter your operating system.
+It is not required, but we recommend using virtual environments such as [`venv`](https://docs.python.org/3/library/venv.html) of [`pipenv`](https://pipenv.readthedocs.io/en/latest/) for repeatable installations that work no matter your operating system.
 
 ```
 $ pipenv --python 3
```

Comment:
This expands the scope a bit - but I think we are still very inconsistent in recommendations across venv, virtualenv and pipenv in various places in docs and product. If we’re sure what’s proposed in these getting started docs is right, we should open an issue to track standardizing on this through everything else.

Reply:
This is definitely an area in which it is possible to get bogged down. There are a variety of tooling options and opinions. If you all have experience-driven and/or Pulumi-specific reasons to recommend `pipenv` that is one thing; if not, I suggest keeping it simple. Something like:

a) we recommend virtual environments for various reasons (should be uncontroversial)

b) you should create and manage your virtual environments in whatever way works best for you, and if you don’t happen to already have a preferred approach then we show you how with the `venv` module of the standard library which you already have

I don’t recall seeing this written down explicitly, but I expect a key reason you are investing in supporting the Pulumi embedded DSL in many host languages is to “meet people where they are”. So most new Pulumi Python users would not be new to the Python ecosystem, and would arrive with a preferred approach already.

(I personally used `pipenv` for awhile, became disillusioned for both technical and other reasons and moved to a combination of `venv` and `pip-tools`. But that is just for context of my comment, not a one-size-fits-all recommendation.)

Reply:
Thanks for all the thoughts and background - definitely helpful!  

We’ve seen folks use a wide variety of tools in this area already with Pulumi, and the good thing of course is that Pulumi itself isn’t really tied to any one of these - it fits well into the existing ecosystem of tools. 

The challenge we’ve faced is more in picking the right “golden path” tooling to use for docs, examples, getting started, installers, etc. We’ve been indecisive on this and ended up suggested different tools at different stages of the user journey. The most important thing at this point is just to be consistent so that users don’t get confused - not really to have any strongly held preference for one tools vs another.  

It is true that many/most users coming to Python + Pulumi already know Python, but I’ve actually been a little surprised at how many folks don’t have a ton of Python experience, and really do rely on these guides to walk though not just the Pulumi aspect, but also handhold through the Python (and cloud) configuration. So we do want to have something simple and clear here. 

Sounds like for now just standardizing on venv is a good simple unopionated built-in option we should just adopt across the board.

Reply:
Will remove.